### PR TITLE
chore(ICP-Rosetta): add spawn neuron functionality

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR proposes the following changes:

1. Add spawn neuron functionality to ICP Rosetta Client
2. Migrate legacy test for spawning new neurons

### CODE REVIEW

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 1]
```diff
@@ -1053,3 +1054,172 @@ fn test_stake_maturity() {
...
+
+                    let new_maturity = 100_000_000_000;
+                    neuron.maturity_e8s_equivalent = new_maturity;
+                    update_neuron(&agent, neuron.into()).await;
+
+                    let neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
+                    assert_eq!(neuron.maturity_e8s_equivalent, new_maturity);
+
+                    let percentage_to_stake = 50;
+                    let spawend_neuron_index = 1;
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:1114]: typo: spaw**ne**d_neuron_index

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 2]
```diff
@@ -1053,3 +1054,172 @@ fn test_stake_maturity() {
...
+                    assert_eq!(neuron.maturity_e8s_equivalent, 0);
+
+                    let new_maturity = 100_000_000_000;
+                    neuron.maturity_e8s_equivalent = new_maturity;
+                    update_neuron(&agent, neuron.into()).await;
+
+                    let neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
+                    assert_eq!(neuron.maturity_e8s_equivalent, new_maturity);
+
+                    let percentage_to_stake = 50;
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:1113]: Should this be `percentage_to_spawn`? Should we also test with an invalid percentage, similar as in `test_stake_maturity`?

Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs]: Done

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs (hunk 1 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs]: Done

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 3]
```diff
@@ -1053,3 +1054,173 @@ fn test_stake_maturity() {
...
+
+                    let neurons = list_neurons(&agent).await.full_neurons;
+                    assert_eq!(neurons.len(), 2);
+                    assert_eq!(
+                        neurons.last().unwrap().controller.unwrap().0,
+                        TEST_IDENTITY.sender().unwrap()
+                    );
+                    assert_eq!(
+                        neurons.last().unwrap().maturity_e8s_equivalent,
+                        new_maturity * percentage_to_stake as u64 / 100 /2
```
Comment (fsodre) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs]: nit: `/ 2` for consistency

**Review by fsodre — APPROVED** [review]

**Pull request merged**